plugins/operators/__init__.py

```diff
@@ -2163,44 +2221,59 @@ def config(self):
     def resolve_input(self, ctx):
         inputs = types.Object()
 
-        workspaces = ctx.dataset.list_workspaces()
-
-        if workspaces:
-            workspace_selector = types.AutocompleteView()
-            for key in workspaces:
-                workspace_selector.add_choice(key, label=key)
-
-            inputs.enum(
-                "name",
-                workspace_selector.values(),
-                default=ctx.spaces.name,
-                required=True,
-                label="Workspace",
-                description="The workspace to delete",
-                view=workspace_selector,
-            )
-        else:
-            prop = inputs.str(
-                "msg",
-                label="This dataset has no saved workspaces",
-                view=types.Warning(),
-            )
-            prop.invalid = True
+        _delete_workspace_inputs(ctx, inputs)
 
         return types.Property(
             inputs, view=types.View(label="Delete workspace")
         )
 
     def execute(self, ctx):
-        name = ctx.params["name"]
+        names = _to_string_list(ctx.params["names"])
 
-        curr_spaces = name == ctx.spaces.name
-        ctx.dataset.delete_workspace(name)
+        curr_spaces = False
+        for name in names:
+            curr_spaces |= name == ctx.spaces.name
+            ctx.dataset.delete_workspace(name)
 
         if curr_spaces:
             ctx.ops.set_spaces(spaces=default_workspace_factory())
 
 
+def _delete_workspace_inputs(ctx, inputs):
+    workspaces = ctx.dataset.list_workspaces()
+
+    if not workspaces:
+        prop = inputs.str(
+            "msg",
+            label="This dataset has no saved workspaces",
+            view=types.Warning(),
+        )
+        prop.invalid = True
+        return
+
+    names = _to_string_list(ctx.params.get("names", []))
+
+    workspace_selector = types.AutocompleteView(multiple=True)
```

Comment:
Ideally can make this `DropdownView(multiple=True)` to avoid manually enforcing valid values below